balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -58,6 +63,20 @@ func newTestConfigBalancerBuilder() *testConfigBalancerBuilder {
 	}
 }
 
+// pickAndCheckError returns a function which takes a picker, invokes the Pick() method
+// multiple times and ensures that the error returned by the picker matches the provided error.
+func pickAndCheckError(want error) func(balancer.Picker) error {
+	const rpcCount = 5
+	return func(p balancer.Picker) error {
+		for i := 0; i < rpcCount; i++ {
+			if _, err := p.Pick(balancer.PickInfo{}); err == nil || !strings.Contains(err.Error(), want.Error()) {
+				return fmt.Errorf("got %q, want error to contain %q, ", want, err)
```

Comment:
Mention that the error was returned from `Pick`. See go/go-style/decisions#identify-the-function

```golang
return fmt.Errorf("Pick() returned error: %v, want: %v", err, want)
```

We usually use `%v` as the formatting directive for error types. If you feel strongly about using %q, then I don't mind. But otherwise, it would be better to stick to `%v`.

Also note that you have for the order or got and want wrong in the arguments list passed to `fmt.Error`.

Reply:
thats a good catch. updating the error message and also flipping the args around

Reply:
thanks for catching that. updating this